beam-postgres/Database/Beam/Postgres/Migrate.hs

```diff
@@ -57,6 +57,7 @@ import           Data.Maybe
 import           Data.ByteString (ByteString)
 import qualified Data.ByteString.Lazy as BL
 import qualified Data.ByteString.Lazy.Char8 as BCL
+import           Data.Monoid (Endo(..))
```

Comment:
`Endo` and `appEndo` are imported via `Data.Semigroup`, so because of the CPP they are not in scope when compiling under GHC 8.4 (it uses `base` >= 4.11).  Thus they have to be explicitly imported for this module to compile.  For older GHC versions this import is just redundant, and has no effect.

Reply:
I see... Can you put a C preprocessor guard around it, and ensure that the sources compile with the -Werror flag on both 8.2 and 8.4?